packages/server/tests/integration/v1/auth/password.spec.ts

```diff
@@ -0,0 +1,192 @@
+    const { user } = await createUser();
+
+    const secretKey = process.env.LOGCHIMP_SECRET_KEY || "test_secret";
+    const payload = { email: user.email, type: "resetPassword" };
+    const token = jwt.sign(payload, secretKey, { expiresIn: "15m" });
+
+    await database("resetPassword").insert({
+      email: user.email,
+      token,
+      createdAt: new Date(),
+    });
+
+    const response = await supertest(app)
+      .post("/api/v1/auth/password/validateToken")
+      .send({ token });
+
+    expect(response.status).toBe(200);
+    expect(response.body.reset.valid).toBe(true);
+    expect(response.body.reset.email).toBe(user.email);
+    expect(response.body.reset.token).toBe(token);
+  });
+});
+
+describe("POST /api/v1/password/set", () => {
+  it("should throw error MISSING_TOKEN if token is missing", async () => {
+    const response = await supertest(app).post(
+      "/api/v1/auth/password/validateToken",
+    );
+
+    expect(response.headers["content-type"]).toContain("application/json");
+    expect(response.status).toBe(400);
+    expect(response.body.errors[0].code).toBe("MISSING_TOKEN");
+  });
+
+  it("should throw error INVALID_TOKEN", async () => {
+    const tokenPayload = {
+      userId: faker.string.uuid(),
+      email: faker.internet.email().toLowerCase(),
+      type: "resetPassword",
+    };
+
+    const token = createToken(tokenPayload, {
+      expiresIn: "2h",
+    });
+
+    const response = await supertest(app)
+      .post("/api/v1/auth/password/validateToken")
+      .send({
+        token,
+      });
+
+    expect(response.headers["content-type"]).toContain("application/json");
+    expect(response.status).toBe(404);
+    expect(response.body.code).toBe("INVALID_TOKEN");
+  });
+
+  it("should throw PASSWORD_MISSING", async () => {
+    const { user } = await createUser();
+
+    const secret = process.env.LOGCHIMP_SECRET_KEY || "test_secret";
+    const token = jwt.sign(
+      { email: user.email, type: "resetPassword" },
+      secret,
+      { expiresIn: "15m" },
+    );
+
+    await database("resetPassword").insert({
+      email: user.email,
+      token,
+      createdAt: new Date(),
+    });
+
+    const response = await supertest(app)
+      .post("/api/v1/auth/password/set")
+      .send({
+        token,
+        password: "",
+      });
+
+    expect(response.status).toBe(400);
+    expect(response.headers["content-type"]).toContain("application/json");
+    expect(response.body.errors[0].code).toBe("PASSWORD_MISSING");
+  });
+
+  it("should successfully reset password and return 200", async () => {
+    const { user } = await createUser();
+
+    const secret = process.env.LOGCHIMP_SECRET_KEY || "test_secret";
+    const token = jwt.sign(
+      { email: user.email, type: "resetPassword" },
+      secret,
+      { expiresIn: "15m" },
+    );
+
+    await database("resetPassword").insert({
+      email: user.email,
+      token,
+      createdAt: new Date(),
+    });
```

Comment:
Instead of this, can you create a util function which basically calls the `/api/v1/auth/password/reset` API endpoint.